core/src/test/kotlin/dev/hotwire/core/bridge/MessageTest.kt

```diff
@@ -48,6 +48,40 @@ class MessageTest {
         assertNull(data)
     }
 
+    @Test
+    fun dataWithNullableFieldsDecodesToObject() {
+        val metadata = Metadata("https://37signals.com")
+        val message = Message(
+            id = "1",
+            component = "page",
+            event = "connect",
+            metadata = metadata,
+            jsonData = """{"title":null,"}""" // subtitle missing
+        )
+
+        val data = message.data<NullableMessageData>()
+
+        assertEquals(null, data?.title)
+        assertEquals(null, data?.subtitle)
```

Comment:
Just a small addition, it'd be helpful to also verity that `data` is not null first to confirm that the object was actually decoded:
```kotlin
assertNotNull(data)
```